### DISCUSSION TITLE
How to refresh a provider

### DISCUSSION THREAD

**Discussion by qu0cquyen** [rrousselGit/riverpod#890] [Q&A]
Hi @rrousselGit, 

First of all thank you so much for such a great library regards to state management. I've been working with it and love it. 

But, I come across a question today while working on my project. 

Is there any possible way that we can automatically re-activate the provider after coming back from a screen ?
For example: 
- Screen A pushes to Screen B
- Screen B will have its own provider and watch all the changes. Then Screen C will be pushed from Screen B. 

How can we automatically refresh the provider of Screen B when we come back from Screen C (Pop) and without the help of callback "context.refresh([provider])". 

To sum up my problem is I expect that there is some sort of function which will trigger the Provider B to be refresh regardless of pushing / popping from the other screens. Hope this clear. 😅😅😅

**Comment by rrousselGit** [comment]
Using `autoDispose` typically would do the trick
